From 48ebd7a71df9a0ad6a99b92045fac33ef0b9b325 Mon Sep 17 00:00:00 2001
From: cpchiang <cpchiang1@nuvoton.com>
Date: Tue, 5 Dec 2023 11:34:08 +0800
Subject: [PATCH 57/71] i3c: master: svc: remove wait i3c bus idle after emit
 stop

SDA may hold by slave immediately after bus become idle,
remove wait to increase i3c bus usage after emit stop.

Signed-off-by: cpchiang <cpchiang1@nuvoton.com>
---
 drivers/i3c/master/svc-i3c-master.c | 6 +-----
 1 file changed, 1 insertion(+), 5 deletions(-)

diff --git a/drivers/i3c/master/svc-i3c-master.c b/drivers/i3c/master/svc-i3c-master.c
index c35720fc191e..6d62ea1a8c0a 100644
--- a/drivers/i3c/master/svc-i3c-master.c
+++ b/drivers/i3c/master/svc-i3c-master.c
@@ -1447,13 +1447,9 @@ static int svc_i3c_master_xfer(struct svc_i3c_master *master,
 		}
 	}
 
-	if (!continued) {
+	if (!continued)
 		svc_i3c_master_emit_stop(master);
 
-		/* Wait idle if stop is sent. */
-		readl_poll_timeout(master->regs + SVC_I3C_MSTATUS, reg,
-				   SVC_I3C_MSTATUS_STATE_IDLE(reg), 0, 1000);
-	}
 	if (!use_dma)
 		local_irq_enable();
 
-- 
2.34.1

